airflow/models.py

```diff
@@ -3978,6 +3983,11 @@ def update_state(self, session=None):
         """
 
         dag = self.get_dag()
+
+        # backfills create subsets of dags with the same name
```

Comment:
And the update_state can't work without this. "update_state" itself doesn't remove task instances, but it needs the whole DAG to assess state, which in turn will be communicated to the user on the frontend.